packages/core/src/Stores/client-store.js

```diff
@@ -747,6 +747,52 @@ export default class ClientStore extends BaseStore {
         }
     }
 
+    @action.bound
+    redirectToMt5Real = (market_type, server, redirectToMt5) => {
+        const synthetic_server_region = server ? server.server_info?.geolocation?.region : '';
+        const synthetic_server_environment = server ? (server.server_info?.environment).toLowerCase() : '';
+
+        const serverElementName = () => {
+            let filter_server_number = '';
+            let synthetic_region_string = '';
+            if (market_type === 'synthetic') {
+                filter_server_number = synthetic_server_environment.split('server')[1];
+                synthetic_region_string = `-${synthetic_server_region.toLowerCase()}`;
+            }
+            return `-${market_type}${synthetic_region_string}${filter_server_number}`;
+        };
+        redirectToMt5(`real${market_type ? serverElementName() : ''}`)
+    };
+
+    @action.bound
+    openMt5RealAccount(account_type, redirectToMt5) {
+        const hasRequiredCredentials = () => {
+            // for MT5 Real Financial STP, if true, users can instantly create a new account by setting password
+            if (!this.account_settings) return false;
+            const { citizen, tax_identification_number, tax_residence } = this.account_settings;
+            return !!(citizen && tax_identification_number && tax_residence);
+        };
+
+        const has_required_account = account_type === 'synthetic' ? this.has_malta_account : this.has_maltainvest_account;
+        const should_redirect_fstp_password = this.is_fully_authenticated && hasRequiredCredentials();
+
+        if (this.is_eu && !has_required_account) {
+            sessionStorage.setItem('cfd_account_needed', 1);
+            this.root_store.ui.toggleAccountsDialog(false);
+            // TO DO remove account needed modal logic
```

Comment:
is it intended to be here?

Reply:
Just left it here to future improvement. Can be removed.